src/seer_utils.erl

```diff
@@ -0,0 +1,72 @@
+-module(seer_utils).
+
+-include("seer.hrl").
+
+-export([carbon_format/1]).
+
+carbon_format(Metrics) ->
```

Comment:
Nit: It would probably be more efficient to generate a fun here that accepts `Name` here and returns the binary expected from `metrics_key/1`. This would avoid the 2 `application:` lookups for every metric and instead have 2 lookups for the whole polling period.

```erlang
Prefix = ?ENV(?ENV_PREFIX, ?DEFAULT_PREFIX),
Host = ?ENV(?ENV_HOST, ?DEFAULT_HOST),
NameFun = fun(Name) ->
    <<Prefix/binary, ".", Host/binary, ".", Name/binary>>
end,
```

Reply:
Even better: You can put Prefix and Host in `seer_server`'s `#state` on initialization and pass them to `carbon_format` as arguments. This would mean that `seer_server` would have to be restarted in order to take config changes into account, but maybe that's not a bad thing.